src/bt_peer_connection.cpp

```diff
@@ -358,6 +358,48 @@ namespace libtorrent {
 		stats_counters().inc_stats_counter(counters::num_outgoing_dht_port);
 	}
 
+	void bt_peer_connection::send_piece_message(message_type const type
```

Comment:
as far as I can tell, all callers of this function also asserts or check: ``TORRENT_ASSERT(m_sent_handshake);`` and ``TORRENT_ASSERT(m_sent_bitfield);`` too. I believe you could factor those into this function.